File: CArticulo/CArticulo/Articulo.cs
using System;
namespace Serpis.Ad.Ventas
{
	public class Articulo
    {
        private decimal precio;
        private ulong? categoria;
        private ulong id;
        private string nombre;

        public ulong Id
        {
            get { return id; }
            set { id = value; }
        }

        public string Nombre
        {
            get { return nombre; }
            set { nombre = value; }
        }
        public ulong? Categoria
        {
            get { return categoria; }
            set { categoria = value; }
        }

        public decimal Precio
        {
            get { return precio; }
            set { precio = value; }
        }
    }
}

File: HPedido/src/main/java/serpis/ad/Articulo.java
package serpis.ad;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;


@Entity(name = "Articulo")
public class Articulo {

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	Long id;
	String nombre;
	double precio;
	
	@ManyToOne
	@JoinColumn(name = "categoria", 
			foreignKey = ForeignKey(name = "CATEGORIA_ID_FK"))
	Categoria categoria;
}
